File: Data/dataset_676/example_code/display_subj.py
# -*- coding: utf-8 -*-
"""
(C) 2018 Wanting Huang <172258368@qq.com>
(C) 2018 Bernd Porr <bernd.porr@glasgow.ac.uk>

GNU GENERAL PUBLIC LICENSE
Version 3, 29 June 2007

Demo program which plots one experiment (Temple Run) from
subject 20. The plot shows the 3 channels recorded: Fp1,
EMG at the chin and the switch which acts as an electronic
clapper board.

"""
import numpy as np
import matplotlib.pyplot as plt
import scipy.signal as signal
import math as math
from eegemg import EegEmg



# plots one experiment from one subject into a single plot window by
# adding a 0.5mV offset to the 2 channels to separate them.
def plotData(subj, experiment, t, fp1, chin, trigger):
    yoffset = 0.001
    plt.title("Subj: "+str(subj)+", Exp:"+experiment)
    plt.xlabel("time(s)")
    plt.ylabel("Amplitude in V")
    plt.plot(t, trigger/5000)
    plt.plot(t, fp1+yoffset/2)
    plt.plot(t, chin-yoffset/2)

        

# main program
# plots subject 20 which is also the example subject in the demo video clip
# playing temple run. Clearly the EOG artefacts can be seen while the subject is doing
# rapid eye movements. The jaw/facial muscle activity is pretty tense as well.
# The 3rd channel is the switch held into the video frame as an electronic clapper board.
# In order to separate the traces and avoiding "subplot" a shift has been introduced.
# If you want to plot more experiments and/or subjects just create subplots in a loop.
#
#
pathToData = "../experiment_data"
#
# subject
subject_number = 20
#
# experiment Temple Run (video game)
experiment = "templerun"
#
# creating class which loads the experiment
eegemg = EegEmg(pathToData, subject_number, experiment)
print("All experiments available? ",eegemg.checkIfSubjectHasAllExperiments(pathToData, subject_number))
#
# filtering out interference
eegemg.filterData()
#
# plot it: you see three traces: the "EEG" from Fp1, the EMG from the chin and the switch
# acting as a "clapper board" to be able to sync video & data.
plotData(subject_number, experiment, eegemg.t, eegemg.fp1, eegemg.chin, eegemg.trigger)
#
# show the experiment
plt.show()

File: Data/dataset_676/example_code/display_subj.m
% (C) 2018 Wanting Huang <172258368@qq.com>
% (C) Bernd Porr <bernd.porr@glasgow.ac.uk>
% 
% GNU GENERAL PUBLIC LICENSE
% Version 3, 29 June 2007
% 
% Demo program which plots one experiment (Temple Run) from
% subject 20. The plot shows the 3 channels recorded: Fp1,
% EMG at the chin and the switch which acts as an electronic
% clapper board.
	 
% only for octave
pkg load signal

% loads data from subject 20 and the experiment templerun
mydata = load("-ascii","../experiment_data/subj20/templerun/emgeeg.dat");

% change in case you need just some part of it
first_sample = 1;
last_sample = size(mydata)(1);

% scaling by the gain of the pre-amp
fp1 = mydata(first_sample:last_sample,2)/500;
chin = mydata(first_sample:last_sample,3)/500;
trigger = mydata(first_sample:last_sample,4);

% sampling rate
fs = 1000;

% filters
[bfilt25hz afilt25hz] = butter(2,[24/fs*2 26/fs*2],'stop');
[bfilt50hz afilt50hz] = butter(2,[49/fs*2 51/fs*2],'stop');
[bfilt80hz afilt80hz] = butter(2,[78/fs*2 82/fs*2],'stop');
[bfilt100hz afilt100hz] = butter(2,[95/fs*2 105/fs*2],'stop');
[bhp ahp] = butter(4,0.5/fs*2,'high');

fp1 = filter(bhp,ahp,filter(bfilt50hz,afilt50hz,fp1));
chin = filter(bhp,ahp,filter(bfilt50hz,afilt50hz,chin));

fp1 = filter(bfilt80hz,afilt80hz,fp1);
chin = filter(bfilt80hz,afilt80hz,chin);

fp1 = filter(bfilt100hz,afilt100hz,fp1);
chin = filter(bfilt100hz,afilt100hz,chin);

% plotting the data
t = linspace(0,last_sample/fs,last_sample);
offset = 0.001;
plot(t,fp1+offset/2,t,chin-offset/2,t,trigger/5000);
xlabel("t/sec");
ylabel("V/Fp1 & chin & trigger");
pause()
